crates/bevy_ecs/src/query/filter.rs

```diff
@@ -490,7 +496,8 @@ macro_rules! impl_or_query_filter {
             }
         }
 
-        impl<$($filter: QueryFilter),*> QueryFilter for Or<($($filter,)*)> {
+            // SAFETY: This only performs access that subqueries perform, and they impl `QueryFilter` and so perform no mutable access.
+            unsafe impl<$($filter: QueryFilter),*> QueryFilter for Or<($($filter,)*)> {
```

Comment:
Same here

Reply:
I'm happy to update the safety comments if they're unclear, but I'm not quite sure what you're proposing to write.  

I was thinking of it like the inductive step of a proof.  We need to prove the safety invariants of `Or<(T, U)> as QueryFilter` are correct, and we do that by relying on the safety invariants of `T as QueryFilter` and `U as QueryFilter` that were proven elsewhere with an `unsafe impl`.

Reply:
I'm not sure what the precedent here is, but I like the style of the current PR.